kolibri/plugins/management/assets/src/views/user-page/edit-user-modal.vue

```diff
@@ -0,0 +1,219 @@
+        :invalid="usernameIsInvalid"
+        :invalidText="usernameIsInvalidText"
+        @blur="usernameBlurred = true"
+        v-model="newUsername"
+      />
+
+      <div>
+        <label for="user-role">
+          <span class="visuallyhidden">{{$tr('userKind')}}</span>
+        </label>
+        <select v-model="newKind" id="user-role">
+          <option :value="LEARNER"> {{$tr('learner')}} </option>
+          <option :value="COACH"> {{$tr('coach')}} </option>
+          <option :value="ADMIN"> {{$tr('admin')}} </option>
+        </select>
+      </div>
+
+      <div class="ta-r">
+        <k-button
+          :text="$tr('cancel')"
+          :primary="false"
+          :raised="false"
+          @click="displayModal(false)"
+        />
+        <k-button
+          type="submit"
+          :text="$tr('save')"
+          :primary="true"
+          :raised="true"
+          :disabled="isBusy"
+        />
+      </div>
+    </form>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import { updateUser, displayModal } from '../../state/actions';
+  import { UserKinds } from 'kolibri.coreVue.vuex.constants';
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import kTextbox from 'kolibri.coreVue.components.kTextbox';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import uiAlert from 'keen-ui/src/UiAlert';
+
+  export default {
+    name: 'editUserModal',
+    $trs: {
+      editUser: 'Edit user',
+      fullName: 'Full name',
+      username: 'Username',
+      userKind: 'User kind',
+      admin: 'Admin',
+      coach: 'Coach',
+      learner: 'Learner',
+      save: 'Save',
+      cancel: 'Cancel',
+      required: 'This field is required',
+      usernameNotAlphaNumUnderscore: 'Username can only contain letters, numbers, and underscores',
+    },
+    components: {
+      kButton,
+      coreModal,
+      kTextbox,
+      uiAlert,
+    },
+    props: {
+      id: {
+        type: String,
+        required: true,
+      },
+      name: {
+        type: String,
+        required: true,
+      },
+      username: {
+        type: String,
+        required: true,
+      },
+      kind: {
+        type: String,
+        required: true,
+      },
+    },
+    data() {
+      return {
+        newName: this.name,
+        newUsername: this.username,
+        newKind: this.kind,
+        nameBlurred: false,
+        usernameBlurred: false,
+        formSubmitted: false,
+      };
+    },
+    computed: {
+      LEARNER: () => UserKinds.LEARNER,
+      COACH: () => UserKinds.COACH,
+      ADMIN: () => UserKinds.ADMIN,
+      nameIsInvalidText() {
+        if (this.nameBlurred || this.formSubmitted) {
+          if (this.newName === '') {
+            return this.$tr('required');
+          }
+        }
+        return '';
+      },
+      nameIsInvalid() {
+        return !!this.nameIsInvalidText;
+      },
+      usernameIsAlphaNumUnderscore() {
+        return /^\w+$/g.test(this.newUsername);
```

Comment:
Should replace with anti-punctuation regex, or maybe turn that regex into a core utility function to import here.

Reply:
Created a core util called `validators` which includes this validation check.